server/images.go

```diff
@@ -839,7 +841,9 @@ func PushModel(ctx context.Context, name string, regOpts *registryOptions, fn fu
 
 	var layers []*Layer
 	layers = append(layers, manifest.Layers...)
-	layers = append(layers, manifest.Config)
+	if manifest.Config.Digest != "" {
```

Comment:
this is a case similar to below where keeping `manifest.Config` a pointer seems more idiomatic

Reply:
in this case, an invalid config should probably not be pushed

Reply:
If you imagine in the future that we change the manifest format so that the config fields are folded in directly to the main file then this situation is not an invalid config.

With this code, an old client would be able to pull a model based on a new manifest and push it, albeit with a few fields missing (the same ones that it wouldn't be able to display, such as number of parameters). This is basically the same behavior for any new fields that we might add.